queue_processors/queue_processor/handle_message.py

```diff
@@ -0,0 +1,388 @@
+        db_access.save_record(data_location)
+
+        # We now need to create all of the variables for the run such that
+        # the script can run
+        # through in the desired way
+        self._logger.info('Creating variables for run')
+        variables = self._utils.instrument_variable.create_variables_for_run(
+            reduction_run)
+        if not variables:
+            self._logger.warning(
+                "No instrument variables found on %s for run %s",
+                instrument.name, message.run_number)
+
+        self._logger.info('Getting script and arguments')
+        reduction_script, arguments = self._utils.reduction_run. \
+            get_script_and_arguments(reduction_run)
+        message.reduction_script = reduction_script
+        message.reduction_arguments = arguments
+
+        # Make sure the RB number is valid
+        try:
+            validate_rb_num(message.rb_number)
+        except InvalidStateException as ex:
+            self._construct_and_send_skipped(
+                message=message, rb_number=message.rb_number, reason=str(ex))
+            raise ex
+
+        if instrument.is_paused:
+            self._logger.info("Run %s has been skipped",
+                              message.run_number)
+        else:
+            self._client.send_message('/queue/ReductionPending', message)
+            self._logger.info("Run %s ready for reduction",
+                              message.run_number)
+
+    # pylint: disable=too-many-arguments
+    def _create_reduction_run_record(self, experiment, instrument, message,
+                                     run_version, script_text, status):
+        reduction_run = self._data_model.ReductionRun(
+            run_number=message.run_number,
+            run_version=run_version,
+            run_name='',
+            cancel=0,
+            hidden_in_failviewer=0,
+            admin_log='',
+            reduction_log='',
+            created=datetime.datetime.utcnow(),
+            last_updated=datetime.datetime.utcnow(),
+            experiment_id=experiment.id,
+            instrument_id=instrument.id,
+            status_id=status.id,
+            script=script_text,
+            started_by=message.started_by)
+        return reduction_run
+
+    def _get_last_run_version(self, run_no, experiment):
+        """
+        Returns the latest run version for the given run number
+        and experiment combo. If none is found (i.e. a new reduction)
+        -1 is returned to indicate this is a new version
+        """
+        last_run = self._data_model.ReductionRun.objects \
+            .filter(run_number=run_no) \
+            .filter(experiment=experiment) \
+            .order_by('-run_version') \
+            .first()
+
+        # By returning -1 callers can blindly increment the version
+        return last_run.run_version if last_run else -1
+
+    def _get_db_inst(self, instrument_name):
```

Comment:
This name is a little misleading as the function also activates the instrument if required. Perhaps its a tad elongated but can we go for `_get_and_activate_instrument(...)` or similar

Reply:
Also needs a docstring